### PR TITLE
[ENG-2159] Sets Cookie for Registration Detail Ember Page

### PR DESCRIPTION
**Pull request by UdayVarkhedkar**
## Purpose
There are a couple of bugs - one where the registration detail ember page was not provided with the cookie with a status message regarding the registration's approval/disproval and another where status messages set on cookies didn't have `id` fields leading to what looks like a translation error.

## Changes
* Adds `id` fields to status messages related to registrations (approval, embargo, public embargo, withdrawal)
* Sets a cookie with a status message for the RegistrationDetail page on the Ember app

## QA Notes
This can be partially tested through the UI. Follow the re-creation steps for the bug [here](https://openscience.atlassian.net/browse/ENG-2159) and you should see a registration approval message on the registration detail page after a moderator on the registration has clicked through the approval link. The message will still display mention of a translation issue because front-end changes are necessary to take advantage of the ids added in this PR. Note, this PR incorporates the backend changes for ENG-2160.

## Documentation
N/A

## Side Effects
N/A

## Ticket
[JIRA Ticket](https://openscience.atlassian.net/browse/ENG-2159)

### CODE REVIEW

**Comment by Johnetordoff** [comment]
Very good! Nice tests!

**Comment by brianjgeiger** [comment]
@UdayVarkhedkar Is this up-to-date with develop? Looks like those tests are still failing.

**Comment by UdayVarkhedkar** [comment]
I think (h/t to Abram) that the Travis build is failing because the test relies on making a request to the ember app which it seems is not a part of the Travis environment. I'm looking into if we can effectively mock the requests to the ember app so the test can still work in Travis.

**Comment by UdayVarkhedkar** [comment]
> @UdayVarkhedkar Is this up-to-date with develop? Looks like those tests are still failing.

It is up to date with develop! I pulled develop and rebased this branch upon my local develop before asking you to restart Travis. All of the tests are passing locally so I think it's what I mentioned in my previous comment.

**Comment by UdayVarkhedkar** [comment]
On Friday, after looking into why the test was failing in Travis (the lack of an ember-osf-web service) and trying to address that, I realized that mocking `set_cookie` wasn't actually an effective test because it is called in a process unrelated to what I was trying to test. In continuing to look into testing the status messages being written properly, and the `osf_status` being set, Abram and I got a point where we concluded that this likely could not be tested because it does not look like we can maintain session stability such that the session that the status message is pushed to is the same as the session the status message is popped from, even when trying to incorporate both into a single, albeit redirected, request. 

In short, in order to test these changes effectively, we need to be able to retrieve the status message pushed upon events like registration approvals from the same session. It does not seem like we are able to do so, at least in a straightforward way, in our current testing environment. I've tested the changes multiple times locally without any issues and think, at this point, this PR continuing without tests would be best.

File: osf/utils/tokens/handlers.py [hunk 1]
```diff
@@ -12,34 +12,66 @@
...
-        'reject': 'Your disapproval has been accepted and the registration has been cancelled.',
-    }[action], kind='success', trust=False)
+    status.push_status_message(
+        message={
+            'approve': 'Your registration approval has been accepted.',
+            'reject': 'Your disapproval has been accepted and the registration has been cancelled.',
+        }[action],
+        id={
+            'approve': 'registration_approved',
+            'reject': 'rgistration_rejected'
```
Comment (corbinSanders) [inline osf/utils/tokens/handlers.py:22]: Typo.

File: osf/utils/tokens/handlers.py [hunk 2]
```diff
@@ -12,34 +12,66 @@
...
-        'reject': 'Your disapproval has been accepted and the embargo has been cancelled.',
-    }[action], kind='success', trust=False)
+    status.push_status_message(
+        message={
+            'approve': 'Your embargo approval has been accepted.',
+            'reject': 'Your disapproval has been accepted and the embargo has been cancelled.',
+        }[action],
+        id={
+            'approve': 'embargo_approved',
+            'reject': 'emargo_rejected'
```
Comment (corbinSanders) [inline osf/utils/tokens/handlers.py:38]: Another typo

File: osf/utils/tokens/handlers.py [hunk 3]
```diff
@@ -12,34 +12,66 @@
 
 def registration_approval_handler(action, registration, registered_from):
     # TODO: Unnecessary and duplicated dictionary.
```
Comment (corbinSanders) [inline osf/utils/tokens/handlers.py:14]: Can we do anything about this TODO?

**Review by corbinSanders — CHANGES_REQUESTED** [review]
Looks good, just see if we can address the TODO while we're here. If we can't, thats ok!

File: osf/utils/tokens/handlers.py (hunk 3 above)
Comment (UdayVarkhedkar) [reply osf/utils/tokens/handlers.py:14]: I can't find the values (status messages) in that dictionary aren't anywhere else in osf.io (I tried to grep) and based on the git blame, the TODO is 5 years old so it seems outdated and I can remove it.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
It looks good, but there are still a number of failing tests.

```
___ TestRegistrationViews.test_register_template_page_backwards_comptability ___
3668tests/test_registrations/test_views.py:100: in test_register_template_page_backwards_comptability
3669    assert_equal(res.status_code, http_status.HTTP_200_OK)
3670E   AssertionError: 302 != 200
3671__________ TestDraftRegistrationViews.test_delete_draft_registration ___________
3672tests/test_registrations/test_views.py:358: in test_delete_draft_registration
3673    assert_equal(1, DraftRegistration.objects.filter(deleted__isnull=True).count())
3674E   AssertionError: 1 != 2
3675 TestDraftRegistrationViews.test_delete_draft_registration_approved_and_registration_deleted 
3676tests/test_registrations/test_views.py:391: in test_delete_draft_registration_approved_and_registration_deleted
3677    assert_equal(1, DraftRegistration.objects.filter(deleted__isnull=True).count())
3678E   AssertionError: 1 != 2
3679_____ TestDraftRegistrationViews.test_delete_draft_registration_non_admin ______
3680tests/test_registrations/test_views.py:366: in test_delete_draft_registration_non_admin
3681    assert_equal(1, DraftRegistration.objects.filter(deleted__isnull=True).count())
3682E   AssertionError: 1 != 2
3683______ TestDraftRegistrationViews.test_only_admin_can_delete_registration ______
3684tests/test_registrations/test_views.py:400: in test_only_admin_can_delete_registration
3685    assert_equal(1, DraftRegistration.objects.filter(deleted__isnull=True).count())
3686E   AssertionError: 1 != 2
```

**Review by brianjgeiger — APPROVED** [review]
